backend/app/controllers/stats_controller.rb

```diff
@@ -6,12 +8,25 @@ def show
     # - there was a unique constraint that does not allow multiple event of the same type
     #   on the same enrollments. Consequently, some submit and request_changes
     #   events are missing in production database.
+    target_api_list = []
 
-    do_filter_by_target_api = params.permit(:target_api).key?(:target_api)
-    target_api = params.permit(:target_api)[:target_api]
-    filter_by_target_api_criteria = do_filter_by_target_api ?
-      "target_api = '#{ActiveRecord::Base.connection.quote_string(target_api)}'" :
-      "1 = 1" # equivalent to no filter
+    if params.permit(:target_api_list).key?(:target_api_list)
+      begin
+        target_api_list = JSON.parse(params[:target_api_list])
```

Comment:
either access target_api_list with "params.permit(:target_api_list)" or with "params[:target_api_list]" but you should be consistent with L13